LTC_extracts_DRAFT.R

```diff
@@ -0,0 +1,137 @@
+
+## packages ##
+
+library(dplyr)
+library(stringr)
+
+
+## financial year in question ##
+FY <- 1718
+year <- convert_fyyear_to_year(FY)
+
+
+## Read data ##
+ltc_file <- readr::read_csv(
+  file = get_it_ltc_path(),
+  col_type = vroom::cols(
+    `PATIENT_UPI [C]` = col_character(),
+    `PATIENT_POSTCODE [C]` = col_character(),
+    `ARTHRITIS_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `ASTHMA_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `ATRIAL_FIB_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `CANCER_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `CEREBROVASC_DIS_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `CHRON_LIVER_DIS_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `COPD_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `DEMENTIA_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `EPILEPSY_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `HEART_DISEASE_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `MULT_SCLEROSIS_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `PARKINSONS_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `RENAL_FAILURE_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `CONGENITAL_PROB_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `BLOOD_AND_BFO_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `OTH_DIS_END_MET_DIAG_DATE` = col_date(format = "%Y/%m/%d %T"),
+    `OTH_DIS_DIG_SYS_DIAG_DATE` = col_date(format = "%Y/%m/%d %T")
+  )
+)
+
+names(ltc_file) <- str_replace_all(names(ltc_file), " ", "_")
+
+## rename ##
+ltc_file <- ltc_file %>%
+  rename(
+    chi = "PATIENT_UPI_[C]",
+    postcode = "PATIENT_POSTCODE_[C]",
+    arth_date = "ARTHRITIS_DIAG_DATE",
+    asthma_date = "ASTHMA_DIAG_DATE",
+    atrialfib_date = "ATRIAL_FIB_DIAG_DATE",
+    cancer_date = "CANCER_DIAG_DATE",
+    cvd_date = "CEREBROVASC_DIS_DIAG_DATE",
+    liver_date = "CHRON_LIVER_DIS_DIAG_DATE",
+    copd_date = "COPD_DIAG_DATE",
+    dementia_date = "DEMENTIA_DIAG_DATE",
+    diabetes_date = "DIABETES_DIAG_DATE",
+    epilepsy_date = "EPILEPSY_DIAG_DATE",
+    chd_date = "HEART_DISEASE_DIAG_DATE",
+    hefailure_date = "HEART_FAILURE_DIAG_DATE",
+    ms_date = "MULT_SCLEROSIS_DIAG_DATE",
+    parkinsons_date = "PARKINSONS_DIAG_DATE",
+    refailure_date = "RENAL_FAILURE_DIAG_DATE",
+    congen_date = "CONGENITAL_PROB_DIAG_DATE",
+    bloodbfo_date = "BLOOD_AND_BFO_DIAG_DATE",
+    endomet_date = "OTH_DIS_END_MET_DIAG_DATE",
+    digestive_date = "OTH_DIS_DIG_SYS_DIAG_DATE"
+  )
+
+
+
+## Create LTC flags 1/0 ##
+
+# Create new variables
+data <- data %>%
+  tibble::add_column(
+    arth = NA,
+    asthma = NA,
+    atrialfib = NA,
+    cancer = NA,
+    cvd = NA,
+    liver = NA,
+    copd = NA,
+    dementia = NA,
+    diabetes = NA,
+    epilepsy = NA,
+    chd = NA,
+    hefailure = NA,
+    ms = NA,
+    parkinsons = NA,
+    refailure = NA,
+    congen = NA,
+    bloodbfo = NA,
+    endomet = NA,
+    digestive = NA
+  )
+
```

Comment:
I've removed this and commited the change as the code on line 114 will create the columns so no need to specify here